hudi-common/src/main/java/org/apache/hudi/common/model/HoodieCommitMetadata.java

```diff
@@ -236,6 +241,44 @@ public static <T> T fromJsonString(String jsonStr, Class<T> clazz) throws Except
     return JsonUtils.getObjectMapper().readValue(jsonStr, clazz);
   }
 
+  /**
+   * parse the bytes of deltacommit, and get the base file and the log files belonging to this
+   * provided file group.
+   */
+  // TODO: refactor this method to avoid doing the json tree walking (HUDI-4822).
+  public static Option<Pair<String, List<String>>> getFileSliceForFileGroupFromDeltaCommit(
+      byte[] bytes, HoodieFileGroupId fileGroupId) {
+    String jsonStr = new String(bytes, StandardCharsets.UTF_8);
+    if (jsonStr.isEmpty()) {
+      return Option.empty();
+    }
+
+    try {
+      JsonNode ptToWriteStatsMap = JsonUtils.getObjectMapper().readTree(jsonStr).get("partitionToWriteStats");
```

Comment:
Actually, why do we even need this method? We have the utils to deser the `HoodieCommitMetadata`, right?

Reply:
we don't have any usable method to get the log files and base file from the commit metadata.
so discussed with @xushiyan @prasannarajaperumal, leave a ticket to solve this.

Reply:
@YannByron we might not have the method to get all the log files but we shouldn't be producing new methods for deserializing `HoodieCommitMetadata` if we already have the methods to do so. Let's follow-up on this one.